multiqc/modules/rsem/rsem.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from Cutadapt """
+
+from __future__ import print_function
+import logging
+import re
+from distutils.version import StrictVersion
+from collections import OrderedDict
+
+from multiqc import config
+#from multiqc.plots import linegraph
+from multiqc.plots import bargraph
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+    """
+    Cutadapt module class, parses stderr logs.
+    Also understands logs saved by Trim Galore!
+    (which contain cutadapt logs)
+    """
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='Rsem', anchor='rsem',
+        href='https://deweylab.github.io/RSEM/',
+        info="RSEM (RNA-Seq by Expectation-Maximization) is a software package for"\
+             "estimating gene and isoform expression levels from RNA-Seq data.")
+    
+         # Find and load any featureCounts reports
+        self.rsem_mappable_data = dict()
+        self.rsem_mapped_data = dict()
+        self.rsem_keys  = list()
+        
+        for f in self.find_log_files('rsem'):
+            
+            if f['s_name'] in self.rsem_mapped_data:
+                log.debug("Duplicate sample name found! Overwriting: {}".format(f['s_name']))
+            else :
+                self.rsem_keys.append(f['s_name'])
+            self.rsem_mapped_data[f['s_name']] = self.parse_rsem_report(f)
+            
+            self.add_data_source(f)
+        # Filter to strip out ignored sample names
+        self.rsem_mappable_data = self.ignore_samples(self.rsem_mappable_data)
+        self.rsem_mapped_data = self.ignore_samples(self.rsem_mapped_data)
+
+        if len(self.rsem_mapped_data) == 0  :
+            log.info("Could not find any reports in {}".format(config.analysis_dir))
+            raise UserWarning
+
+        log.info("Found {} reports".format(len(self.rsem_mapped_data)))
+
+        # Write parsed report data to a file
+        self.write_data_file(self.rsem_mapped_data, 'multiqc_mappable_data')
+
+        # Basic Stats Table
+        self.rsem_stats_table()
+
+        # Assignment bar plot
+        self.add_section( plot = self.rsem_chart() )
+
+
+    def parse_rsem_report (self, f):
+        """ Parse the rsem cnt stat file. """
+        data = dict()
+        for l in f['f'].splitlines():
+            s = l.split(" ")
```

Comment:
I think this is basically equivalent to `l.split()` - probably cleaner to not specify the whitespace string.